src/common/MagentoClientBase.js

```diff
@@ -113,13 +117,49 @@ class MagentoClientBase {
             headers: this.headers,
             json: true
         };
+
+        if (this.args.DEBUG) {
+            return this._profileRequest(options);
+        }
         return requestPromise(options);
     }
-    
+
+    /**
+     * Wraps a request and prints out debug information to the log.
+     * 
+     * @param {Object} options  Request options.
+     */
+    _profileRequest(options) {
+        let s = process.hrtime();
+        let logRequest = this._logRequest.bind(this);
```

Comment:
why is this here for?

Reply:
Inside the promise.then, `this` no longer refers to the `MagentoClientBase` class, but the promise context. So you either store the class context inside a different variable (e.g. `that`) or you can store a copy of the function that is already bound to the class context in a different scope and use it from the promise.

Reply:
and where is the class context used in the `_logRequest`? Is the `process.hrTime()` needing it ?

Reply:
No, it would work without the bind as well. If someone would change the logRequest method however and add a `this`, I think he would expect it to work, since the scoping might not be obvious.

Reply:
OK, then please add a simple inline comment to the bind statement.